api/tests/routes/pro/get_suggested_subcategories_test.py

```diff
@@ -0,0 +1,46 @@
+from unittest import mock
+
+import pytest
+
+from pcapi.core import testing
+from pcapi.core.external.subcategory_suggestion_backends import fixtures
+from pcapi.core.external.subcategory_suggestion_backends.subcategory_suggestion import SubcategorySuggestionBackend
+import pcapi.core.offerers.factories as offerers_factories
+import pcapi.core.users.factories as users_factories
+
+
+@pytest.mark.usefixtures("db_session")
+class Returns401Test:
+    def test_access_by_no_one(self, client):
+        response = client.get("/offers/suggested-subcategories?offer_name=foo")
+
+        assert response.status_code == 401
+
+
+@pytest.mark.usefixtures("db_session")
+@mock.patch("pcapi.core.external.subcategory_suggestion.subcategory_suggestion_backend", SubcategorySuggestionBackend())
+@mock.patch("pcapi.core.auth.api.get_id_token_from_google", return_value="Good token")
+class Returns200Test:
+    num_queries = 0
+    num_queries += 1  # get venue
+    num_queries += 1  # get offerer
+    num_queries += 17  # google auth
+
+    def test_suggest_subcategories(self, mock_get_id_token_from_google, requests_mock, client):
+        # Given
+        pro = users_factories.ProFactory()
+        venue = offerers_factories.VenueFactory()
+        requests_mock.post(
+            "https://compliance.passculture.team/latest/model/categorisation",
+            json=fixtures.SUBCATEGORY_SUGGESTION_MUSEE_RESULT,
+        )
+
+        # When
+        with testing.assert_num_queries(self.num_queries):
+            response = client.with_session_auth(email=pro.email).get(
+                f"/offers/suggested-subcategories?offer_name=foo&offer_description=bar&venue_id={venue.id}"
+            )
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏je pense que tu peux t'en sortir mieux, en sortant des choses qui font des queries à cause du test, et pas du code (je pense, à vérifier)

```python
client = client.with_session_auth(email=pro.email) # ça sort du décompte, c'est pour le test
venue_id = venue.id # ça peut parfois être une query
with testing.assert_num_queries(self.num_queries):
    response = client.get(f"/offers/suggested-subcategories?offer_name=foo&offer_description=bar&venue_id={venue_id}")
```